siliconcompiler/core.py

```diff
@@ -492,122 +471,104 @@ def find_function(self, modulename, functype, funcname):
 
-        The target function imports and executes a set of setup functions based
-        on a '_' separated string. The following target string combinations are
-        permitted:
+        The function searches the $SCPATH for targets/<name>.py and runs
+        the setup function in that module if found.
 
-        * <projname>
-        * <flowname>
-        * <flowname>_<pdkname>
-        * <flowname>_<partname> (for fpga flows)
-        * <pdk>
-        * <tool>
-        * <tool>_<pdkname>
+        Args:
+            name (str): Module name
+            flow (str): Target flow to
 
-        If no target name is provided, the target will be read from the
-        'target' schema parameter. Calling target() with no target name provided
-        and an undefined 'target' parameter results in an error.
+        Examples:
+            >>> chip.load_target('freepdk45_demo')
+            Loads the 'freepdk45_demo' target
+
+        """
 
-        The target function uses the find_function() method to import and
-        execute setup functions based on the 'scpath' search parameter.
+        self.set('target', name)
+
+        func = self.find_function(name, 'setup', 'targets')
+        if func is not None:
+            func(self)
+        else:
+            self.logger.error(f'Module {name} not found.')
+            sys.exit(1)
+
+    ##########################################################################
+    def load_pdk(self, name):
+        """
+        Loads a PDK module and runs the setup() function.
 
+        The function searches the $SCPATH for pdks/<name>.py and runs
+        the setup function in that module if found.
 
         Args:
-            name (str): Name of target combination to load.
+            name (str): Module name
 
         Examples:
-            >>> chip.target("asicflow_freepdk45")
-            Loads the 'freepdk45' and 'asicflow' setup functions.
-            >>> chip.target()
-            Loads target based on result from chip.get('target')
+            >>> chip.load_pdk('freepdk45_pdk')
+            Loads the 'freepdk45' pdk
 
         """
 
-        #Sets target in dictionary if string is passed in
-        if name is not None:
-            self.set('target', name)
-
-        # Error checking
-        if not self.get('target'):
-            self.logger.error('Target not defined.')
-            sys.exit(1)
-        elif len(self.get('target').split('_')) > 2:
-            self.logger.error('Target should have zero or one underscore')
+        func = self.find_function(name, 'setup', 'pdks')
+        if func is not None:
+            func(self)
+        else:
+            self.logger.error(f'Module {name} not found.')
             sys.exit(1)
 
-        target = self.get('target')
-        self.logger.info(f"Loading target '{target}'")
-
-        # search for module matches
-        targetlist = target.split('_')
-
-        for i, item in enumerate(targetlist):
```

Comment:
Great that all this logic is going away, it was getting complicated!

Reply:
Yes!